File: .cursor-agent/hooks/afterShellExecution.py
#!/usr/bin/env python3
"""
afterShellExecution Hook
在 Agent 执行 Shell 命令后触发，用于审计和收集指标
"""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent / "lib"))

from agent_hook_handler import AuditHook


class AfterShellExecutionHook(AuditHook):
    """Shell 命令执行后的审计"""
    
    def __init__(self):
        super().__init__("afterShellExecution")
    
    def audit(self) -> None:
        """审计命令执行"""
        command = self.input_data.get("command", "")
        output = self.input_data.get("output", "")
        exit_code = self.input_data.get("exit_code", 0)
        
        if not command:
            return
        
        # 生成简洁的命令预览
        cmd_preview = command[:30] + "..." if len(command) > 30 else command
        
        # 检查输出中的错误
        has_error = exit_code != 0 or any(
            keyword in output.lower() 
            for keyword in ['error', 'failed', 'exception', 'traceback']
        )
        
        if has_error:
            # 命令执行失败
            self.send_to_ortensia(
                f"命令失败：{cmd_preview}",
                emotion="sad"
            )
        else:
            # 命令执行成功（所有命令都通知）
            self.send_to_ortensia(
                f"命令完成：{cmd_preview}",
                emotion="happy"
            )


if __name__ == "__main__":
    hook = AfterShellExecutionHook()
    sys.exit(hook.run())

File: cursor-hooks/lib-node/hooks/afterShellExecution.js
#!/usr/bin/env node
/**
 * afterShellExecution (Node)
 */

const path = require("path");
const { AuditHook } = require(path.join(__dirname, "..", "agent_hook_handler"));

class AfterShellExecutionHook extends AuditHook {
  constructor() {
    super("afterShellExecution");
  }

  async audit() {
    const command = this.inputData.command || "";
    const output = this.inputData.output || "";
    const exitCode = typeof this.inputData.exit_code === "number" ? this.inputData.exit_code : 0;

    if (!command) return;

    const cmdPreview = command.length > 30 ? `${command.slice(0, 30)}...` : command;
    const lowerOut = String(output).toLowerCase();
    const hasError =
      exitCode !== 0 ||
      ["error", "failed", "exception", "traceback"].some((k) => lowerOut.includes(k));

    if (hasError) {
      try {
        await this.sendToOrtensia(`命令失败：${cmdPreview}`, "sad");
      } catch {}
    } else {
      try {
        await this.sendToOrtensia(`命令完成：${cmdPreview}`, "happy");
      } catch {}
    }
  }
}

module.exports = (async () => {
  const code = await new AfterShellExecutionHook().run();
  return code;
})();

